daffodil-test/src/test/resources/org/apache/daffodil/section00/general/testBase.tdml

```diff
@@ -0,0 +1,69 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<tdml:testSuite xmlns:tdml="http://www.ibm.com/xmlns/dfdl/testData"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:ex="http://example.com"
+  xmlns:fn="http://www.w3.org/2005/xpath-functions"
+  xmlns:daf="urn:ogf:dfdl:2013:imp:daffodil.apache.org:2018:ext"
+  suiteName="unparser tests involving bitOrder and outputValueCalc interactions"
+  defaultRoundTrip="false">
+
+  <tdml:defineSchema name="s1" elementFormDefault="unqualified">
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:format ref="ex:GeneralFormat" lengthKind="delimited"/>
+    <xs:element name="foo">
+      <xs:simpleType>
+        <xs:restriction base="xs:string ">
+          <xs:pattern value="&#xE000;*"/>
+        </xs:restriction>
+      </xs:simpleType>
+    </xs:element>
+  </tdml:defineSchema>
+
+  <tdml:defineSchema name="s2" elementFormDefault="unqualified">
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:format ref="ex:GeneralFormat" lengthKind="delimited"/>
+    <xs:element name="foo">
+      <xs:simpleType>
+        <xs:restriction base="xs:string">
+          <xs:pattern value="&#xE000;*"/>
+        </xs:restriction>
+      </xs:simpleType>
+    </xs:element>
+  </tdml:defineSchema>
```

Comment:
It'd be clearer to rename these schemas based on what they're testing. Ex: RestrictionBase.embedded.dfdl.xsd. The embedded.dfdl.xsd is helpful while debugging if you run into issues. I think the file should be changed to testRestrictionBase. testBase is abit ambiguous. I had to go look at the ticket to understand what was being tested.

The elements could also be named in a manner than describes the difference ex foo_noSpaceInAttribute and foo_spaceInAttribute. I had to stare hard to discover the difference

Reply:
I'd also like to see a test for leading space